src/main/scala/fs/fusejna.scala

```diff
@@ -0,0 +1,114 @@
+package suffuse
+package fs
+
+import net.fusejna._
+import jio._
+import scala.util.Properties.isMac
+
+/** Cleaning up the javacentric fuse-jna types.
+ */
+
+object Node {
+  import net.fusejna.types.TypeMode.NodeType
+
+  final val BlockDev = NodeType.BLOCK_DEVICE
+  final val Dir      = NodeType.DIRECTORY
+  final val Fifo     = NodeType.FIFO
+  final val File     = NodeType.FILE
+  final val Link     = NodeType.SYMBOLIC_LINK
+  final val Socket   = NodeType.SOCKET
+}
+
+/** Widening access so we don't have to use inheritance everywhere.
+ */
+
+trait FuseFs extends FuseFilesystem {
+  def unmountTry(): Unit = (
+    if (!isMounted)
+      return
+    else if (isMac)
+      exec("umount", "-f", getMountPoint.getPath) orElse exec("diskutil", "unmount", getMountPoint.getPath)
+    else
+      exec("fusermount", "-u", getMountPoint.getPath)
+  )
+  private def doMount(mountPoint: File, blocking: Boolean): this.type = {
+    addUnmountHook(this)
+    super.mount(mountPoint, blocking)
+    this
+  }
+
+  def mountBg(mountPoint: Path): this.type = doMount(mountPoint.toFile, false)
+  def mountFg(mountPoint: Path): this.type = doMount(mountPoint.toFile, true)
```

Comment:
I'd never use a boolean argument in the first place if I didn't need it for the underlying interface, but I agree. If only I could fucking [label the calls into java](https://issues.scala-lang.org/browse/SI-5896), which is 98% of where I'm going to need it!